File: 27-object-oriented-javascript/movies-frontend/Dog.js
// let neikko = {
//   name: "Neikko",
//   favoriteToy: "rope",
//   speak: function(){
//     return `Woof woof! My name is ${this.name}`
//   }
// }

// let perky = {
//   name: "Perky",
//   favoriteToy: "poop",
//   speak: function(){
//     return `Woof woof!!!! My name is ${this.name}`
//   }
// }

class Dog{
  constructor(name, favoriteToy){
    this.name = name
    this.favoriteToy = favoriteToy
  }

  speak(){
    return `Woof woof! My name is ${this.name}`
  }

  static eat(){
    return "I'm eating!! Tasty"
  }
}

let neikko = new Dog("neikko", "rope")
let perky = new Dog("Perky", "horse poo")

File: 27-object-oriented-javascript/dog.rb
require 'pry'

class Dog
  attr_accessor :name, :favorite_toy
  
  def initialize(name, favorite_toy)
    @name = name
    @favorite_toy = favorite_toy
  end

  def speak
    "Woof woof, my name is #{self.name}"
  end

  def self.walk
    "I'm of class #{self} and I'm walking"
  end
end

neikko = Dog.new("Neikko", "rope")
perky = Dog.new("Perky", "horse manure")

binding.pry
puts 'okthxbye' 